apps/aecore/src/aec_peers.erl

```diff
@@ -60,26 +60,30 @@
 
 %% We parse the uri's with http_uri, therefore we use types from that module.
 -record(peer, {
-          uri = ""          :: http_uri:uri(),   %% try not to use!
+          uri = ""          :: http_uri_uri(),   %% try not to use!
```

Comment:
These are definitely workarounds and OTP http_uri shall be patched, though we still need these workarounds for current OTP. I suggest merging these workarounds and I plan to see how it easy it is to put together patch from OTP.

Reply:
I opened OTP PR 1724 after having tested it makes Dialyzer pass. Though not sure when it will be merged.